NCS.DSS.Address/PatchAddressHttpTrigger/Service/PatchAddressHttpTriggerService.cs

```diff
@@ -2,68 +2,93 @@
 using NCS.DSS.Address.Cosmos.Provider;
 using NCS.DSS.Address.Models;
 using NCS.DSS.Address.ServiceBus;
-using System;
 using System.Net;
-using System.Threading.Tasks;
 
 namespace NCS.DSS.Address.PatchAddressHttpTrigger.Service
 {
     public class PatchAddressHttpTriggerService : IPatchAddressHttpTriggerService
     {
-
-        private readonly IDocumentDBProvider _documentDbProvider;
+        private readonly ICosmosDbProvider _cosmosDbProvider;
         private readonly IAddressPatchService _addressPatchService;
+        private readonly IAddressServiceBusClient _addressServiceBusClient;
+        private readonly ILogger<PatchAddressHttpTriggerService> _logger;
 
-        public PatchAddressHttpTriggerService(IDocumentDBProvider documentDbProvider, IAddressPatchService addressPatchService)
+        public PatchAddressHttpTriggerService(ICosmosDbProvider cosmosDbProvider, IAddressPatchService addressPatchService, IAddressServiceBusClient addressServiceBusClient, ILogger<PatchAddressHttpTriggerService> logger)
         {
-            _documentDbProvider = documentDbProvider;
+            _cosmosDbProvider = cosmosDbProvider;
             _addressPatchService = addressPatchService;
+            _addressServiceBusClient = addressServiceBusClient;
+            _logger = logger;
         }
 
-        public string PatchResource(string addressJson, AddressPatch addressPatch, ILogger logger)
+        public string PatchResource(string addressJson, AddressPatch addressPatch)
         {
-            logger.LogInformation("started patching address");
+            _logger.LogInformation("Started patching address");
             if (string.IsNullOrEmpty(addressJson))
             {
-                logger.LogInformation("Can't patch address because input address json is null");
+                _logger.LogInformation("Can't patch address because input address json is null");
                 return null;
             }
 
             if (addressPatch == null)
             {
-                logger.LogInformation("Can't patch address because input addressPatch object is null");
+                _logger.LogInformation("Can't patch address because input addressPatch object is null");
                 return null;
             }
 
+            _logger.LogInformation("Setting default values for address PATCH object.");
             addressPatch.SetDefaultValues();
-            var addressObj = _addressPatchService.Patch(addressJson, addressPatch, logger);
+            _logger.LogInformation("Default values for address PATCH object are successfully set.");
+
+            var addressObj = _addressPatchService.Patch(addressJson, addressPatch);
 
-            logger.LogInformation("completed patching address");
+            _logger.LogInformation("Completed patching address");
 
             return addressObj;
         }
 
         public async Task<Models.Address> UpdateCosmosAsync(string addressJson, Guid addressId, ILogger logger)
```

Comment:
Can we remove the logger from the params here and use the one in the constructor